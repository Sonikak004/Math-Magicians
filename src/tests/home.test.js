// Home.test.js
import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from '../components/home';
import '@testing-library/jest-dom/extend-expect';

describe('Home component', () => {
  test('renders correctly', () => {
    render(<Home />);
    const homeTextElement = screen.getByText(/Welcome to the Home Page!/i);
    expect(homeTextElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Add more test cases for the Home component if needed
});
