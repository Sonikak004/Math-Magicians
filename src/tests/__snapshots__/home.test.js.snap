// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="home-text"
    >
      <h3>
        Welcome to the Home Page!
      </h3>
      <p>
        The Math Magician website is an interactive platform created using React and CSS, featuring a captivating design with a responsive layout. It combines a fully functional calculator with a random quote generator. The website welcomes users with a delightful image of the Math Magician and encourages exploration of its functionalities. The calculator allows users to perform basic arithmetic operations using on-screen buttons and keyboard inputs. React Router handles seamless navigation between the Home, Calculator, and Quotes pages, while JavaScript logic enables real-time calculations and event handling. 
      </p>
      <br />
      <p>
        The Quotes page fetches random quotes from an external API using the \`fetch\` function in React. Error handling ensures graceful handling of unexpected issues, and accessibility features make the website user-friendly for all. The Math Magician website is a dynamic and engaging platform, providing users with both mathematical assistance and inspirational quotes to brighten their day. 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
